ql/src/java/org/apache/hadoop/hive/ql/optimizer/SimpleFetchOptimizer.java

```diff
@@ -152,28 +152,32 @@ private FetchTask optimize(ParseContext pctx, String alias, TableScanOperator so
   }
 
   private boolean checkThreshold(FetchData data, int limit, ParseContext pctx) throws Exception {
-    if (limit > 0) {
-      if (data.hasOnlyPruningFilter()) {
-        /* partitioned table + query has only pruning filters */
-        return true;
-      } else if (data.isPartitioned() == false && data.isFiltered() == false) {
-        /* unpartitioned table + no filters */
-        return true;
+    boolean cachingEnabled = HiveConf.getBoolVar(pctx.getConf(), HiveConf.ConfVars.HIVEFETCHTASKCACHING);
+    if (!cachingEnabled) {
+      /* if caching is enabled we apply the treshold in all cases */
```

Comment:
The `cachingEnabled` variable name and the comment suggest that true means caching is enabled but the `if` predicate is negated. Does is means that this branch should run when caching is disabled?

Reply:
No, the if construct is fine, I moved this comment outside of the if block. It was  placed poorly.